include/RAJA/policy/cuda/kernel/ForICount.hpp

```diff
@@ -0,0 +1,234 @@
+/*!
+ ******************************************************************************
+ *
+ * \file
+ *
+ * \brief   Header file for CUDA statement executors.
+ *
+ ******************************************************************************
+ */
+
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-18, Lawrence Livermore National Security, LLC.
+//
+// Produced at the Lawrence Livermore National Laboratory
+//
+// LLNL-CODE-689114
+//
+// All rights reserved.
+//
+// This file is part of RAJA.
+//
+// For details about use and distribution, please read RAJA/LICENSE.
+//
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+
+#ifndef RAJA_policy_cuda_kernel_ForICount_HPP
+#define RAJA_policy_cuda_kernel_ForICount_HPP
+
+#include "RAJA/config.hpp"
+
+#include "RAJA/policy/cuda/kernel/internal.hpp"
+
+
+namespace RAJA
+{
+
+namespace internal
+{
+
+
+
+/*
+ * Executor for thread work sharing loop inside CudaKernel.
+ * Mapping directly from threadIdx.xyz to indices
+ * Assigns the loop index to offset ArgumentId
```

Comment:
Should this be thread index instead of loop index for the offset ArgumentId?

Reply:
It's not the thread index... rather it's more like  segment.begin()+threadIdx.?
So I updated the language

Reply:
Thanks.